### ISSUE TITLE
How to use pretrained word embeddings?

### ISSUE THREAD

**Issue by miikargh** [lvapeab/nmt-keras#60]
First of all, thanks for the awesome work you've done here!

I'm trying to build an English-Finnish translator using the pre-defined model but can't figure out how to use custom pre-trained word embeddings. I've looked through the tutorials but didn't find answers there.

All help appreciated :)

**Comment by lvapeab** [comment]
Hi,

I assume you've got your word embeddings with the `binary` or `text` format from `word2vec`, `Glove`, `Fasttext` or a similar one. If so, you must pre-process your word embeddings with the correspoding script from the `utils`dir,  [binary](https://github.com/lvapeab/nmt-keras/blob/master/utils/preprocess_binary_word_vectors.py) or [text](https://github.com/lvapeab/nmt-keras/blob/master/utils/preprocess_text_word_vectors.py). They will be stored as, say, `mydir/awesombeddings.en.npy`. and `mydir/awesombeddings.fi.npy`.

Next, you must modify the `config.py` file, for indicating where are stored your pre-trained word embeddings: 
```
SRC_PRETRAINED_VECTORS = 'mydir/awesombeddings.en.npy'
[....]
TRG_PRETRAINED_VECTORS = 'mydir/awesombeddings.fi.npy'  ```

Hope this help.

Cheers!

**Comment by miikargh** [comment]
Hi!

Thanks for the quick reply!

Ideally, I'd use Fasttext embeddings, since they work best with Finnis words, but the ones downloaded from <a href="https://github.com/facebookresearch/fastText/blob/master/pretrained-vectors.md">here</a> don't seem to be compatible.

I tested the <a href="https://github.com/lvapeab/nmt-keras/blob/master/utils/preprocess_text_word_vectors.py">text</a> script with English Glove embeddings in form from https://nlp.stanford.edu/projects/glove/ and it worked fine.

So Fasttext is not compatible, I gather?

**Comment by lvapeab** [comment]
Hi,

I've just pushed a fix to this (01e4cd25d026dbfdf62ff007abeb497b47619872). I tested it with the Finish text embeddings from Fasttext and seems to work. In few days, I'll test the script for binary embeddings.

Please, reopen this issue if you face other problems. 

Cheers!

**Comment by miikargh** [comment]
First, I preprocessed the embeddings with the corresponding file from the `utils`directory. I used the `preprocess_text_word_vectors.py` with the following command, since I downloaded the text embeddings.
```
python utils/preprocess_text_word_vectors.py -v ~/word_embeddings/finnish/wiki.fi/wiki.fi.vec -d data/word_embeddings/finnish_fasttext.wiki.fi
```
I added some error handling to the `preprocess_text_word_vectors.py` to make it work correctly, but @lvapeab fixed it since, I guess. The changes I made can be seen above.

The script produces a funny filename for some reason, for me it was `data/word_embeddingsfinnish_fasttext.wiki.fi.npy` instead of `data/word_embeddings/finnish_fasttext.wiki.fi.npy` so I manually changed it with command
```
mkdir data/word_embeddings && mv data/word_embeddingsfinnish_fasttext.wiki.fi.npy data/word_embeddings/finnish_fasttext.wiki.fi.npy
```

Second, I changed variables `SRC_PRETRAINED_VECTORS` and `SOURCE_TEXT_EMBEDDING_SIZE` in `config.py` like this:
```python
SOURCE_TEXT_EMBEDDING_SIZE = 300
SRC_PRETRAINED_VECTORS = 'data/word_embeddings/finnish_fasttext.wiki.fi.npy'
```
And finally, I just ran the command `python main.py` and the script started the training.

Hope you find this helpful!
